libafl/src/corpus/inmemory_ondisk.rs

```diff
@@ -304,7 +305,13 @@ where
     fn save_testcase(&self, testcase: &mut Testcase<I>, idx: CorpusId) -> Result<(), Error> {
         let file_name_orig = testcase.filename_mut().take().unwrap_or_else(|| {
             // TODO walk entry metadata to ask for pieces of filename (e.g. :havoc in AFL)
-            testcase.input().as_ref().unwrap().generate_name(idx.0)
+            let mut hasher = ahash::RandomState::with_seeds(0, 0, 0, 0).build_hasher();
```

Comment:
Why always create the hasher, even when it's not used inside the function? Seems wasteful and somewhat pointless?

Reply:
generate_name() should use it.

`generate_name(idx.0, &mut hasher)`

Reply:
It doesn't always? Why not create it inside if you need it? Imho an ideal name shouldn't be random, but log the mutations etc.

Reply:
Nautilus doesn't use it for example

Reply:
The purpose of putting hasher into the argument is to making it clear it is suggested to use hasher to make the testcase names random. 
https://github.com/AFLplusplus/LibAFL/issues/1232#issuecomment-1526153149

> Imho an ideal name shouldn't be random, but log the mutations etc.
Yup, like afl, but not all mutators log the mutation result

Reply:
I don't think that's a good solution, just adding a comment should be better. No need to change the API imho.
That being said, we can also consider having one corpus folder for each core somehow

Reply:
hmm ok

> That being said, we can also consider having one corpus folder for each core somehow
in tlspuffin's case there were 25 testcases with the same name, while there were only 8 cores fuzzing.

I think the problem is https://github.com/AFLplusplus/LibAFL/blob/main/libafl/src/corpus/inmemory_ondisk.rs#L316
this counter. It's easy to have clashing names if you monotonously increase it.

How do you think about putting the current time in nano sec instead of a counter here?

Reply:
that's something that tlspuffin should do